src/datasets/utils/spark_utils.py

```diff
@@ -0,0 +1,35 @@
+import os
+import uuid
+
+import pyspark
+
+
+def validate_cache_dir(cache_dir, spark_function_name):
```

Comment:
Let's add an option to pass in spark so that Spark's self._spark can be reused?

Reply:
Also let's add a comment on this function that also describes what `spark_function_name` is being used for (error message)